concourse/client_test.go

```diff
@@ -2,144 +2,144 @@ package concourse
-		w.WriteHeader(http.StatusUnauthorized)
-	}))
-	defer unauthorized.Close()
+const tokenType = "Bearer"
+const accessToken = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJjc3JmIjoiMDM3YWQ5Zjk5OGMyYmViZDg1MGMzN2NkOTkwMGE2YjdmOTZjOTkwYzk4ZDk3YWQyNjliYTU2N2IyMTI5MjJkZCIsImV4cCI6MTUzMDc0NzA5NiwiaXNBZG1pbiI6dHJ1ZSwidGVhbU5hbWUiOiJtYWluIn0.YL11IVWLlkx5xV_aX7DT0f_Y_FIA8NS5jOndZFw8paJBUVv-_brvTAD9hn4t6FPw0o2gsub1jHB2E4l-VB954_iQ-SfWyzDx0idegYrLlcgsXHhPWku1hW1JBjvq3BhjkKgAuZCW4JP5UulglObaKKFFhYycMZiiiWcKM_zMpn7ebgP6giEemSRj06Bpc5EKAZeZjt0Tv3AqEKd693qI9XJp49LJwZJP_RZgCoXMduLQpm3UmIQppwzFUEyIcAfXJmYvi3utr_JjxfuVuwqZbsemf_fCxRGgkUcwRBwBnlRqvBUdErk63HYAL7t4pdk9mzb61U5OPK9XT8NS195IHw"
 
+func TestNewClient(t *testing.T) {
 	cases := map[string]struct {
 		username string
 		password string
-		apiurl   string
-		team     string
-		want     *Client
-		err      bool
+
+		public bool
+		legacy bool
+		err    bool
 	}{
-		"public pipeline": {
-			apiurl: ok.URL,
-			team:   "main",
-			want: &Client{
-				apiurl: fmt.Sprintf("%s/api/v1", ok.URL),
-				team:   "main",
-			},
+		"public": {
+			public: true,
 		},
-		"private pipeline": {
+		"legacy": {
+			username: "admin",
+			password: "sup3rs3cret1",
+
+			legacy: true,
+		},
+		"skymarshal": {
 			username: "admin",
 			password: "sup3rs3cret1",
-			apiurl:   ok.URL,
-			team:     "main",
-			want: &Client{
-				apiurl: fmt.Sprintf("%s/api/v1", ok.URL),
-				team:   "main",
-				auth:   token,
-			},
 		},
 		"unauthorized": {
 			username: "admin",
 			password: "sup3rs3cret1",
-			apiurl:   unauthorized.URL,
-			team:     "main",
-			err:      true,
+
+			err: true,
 		},
 	}
 
 	for name, c := range cases {
+		s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if c.err {
+				http.Error(w, "", http.StatusUnauthorized)
+			}
+
+			var version string
+			if c.legacy {
+				version = "3.14.2"
+			} else {
+				version = "4.0.0"
+			}
+
+			var resp []byte
+			switch r.RequestURI {
+			case "/api/v1/info":
+				resp, _ = json.Marshal(Info{ATCVersion: version})
+			case "/sky/token":
+				resp, _ = json.Marshal(oauth2.Token{TokenType: tokenType, AccessToken: accessToken})
+			default:
+				resp, _ = json.Marshal(Token{Type: tokenType, Value: accessToken})
+			}
+
+			w.Header().Set("Content-Type", "application/json")
+			w.Write(resp)
+		}))
+
 		t.Run(name, func(t *testing.T) {
-			client, err := NewClient(c.apiurl, c.team, c.username, c.password)
+			client, err := NewClient(s.URL, "main", c.username, c.password)
 			if err != nil && !c.err {
 				t.Fatalf("unexpected error from NewClient:\n\t(ERR): %s", err)
 			} else if err == nil && c.err {
 				t.Fatalf("expected an error from NewClient:\n\t(GOT): nil")
-			} else if !reflect.DeepEqual(client, c.want) {
-				t.Fatalf("unexpected Client from NewClient:\n\t(GOT): %#v\n\t(WNT): %#v", client, c.want)
+			} else if client == nil {
```

Comment:
Wait. Why is the client ever nil? This should probably be checking the length of the cookies to exit early instead.